orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationFilterException.java

```diff
@@ -0,0 +1,56 @@
+package com.sap.ai.sdk.orchestration;
+
+import com.google.common.annotations.Beta;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.experimental.StandardException;
+
+/**
+ * Exception thrown when an error occurs during orchestration filtering.
+ *
+ * <p>This exception serves as the base for more specific filter-related exceptions in the
+ * orchestration process.
+ */
+@Beta
+@StandardException(access = AccessLevel.PRIVATE)
+public class OrchestrationFilterException extends OrchestrationClientException {
+
+  /** Details about the filter that caused the exception. */
+  @Getter @Nonnull protected Map<String, Object> filterDetails;
```

Comment:
(Minor)

IntelliJ is correctly warning me: This field cannot be "Nonnull" when default constructor of this class sets it to "null". I think there are multiple ways to address that.

Reply:
Easiest (and obvious?) way is probably initializing it with empty map.